Scenarios/Secure-Baseline/bicepWithAVM/03-Supporting-Services/modules/key-vault/types.bicep

```diff
@@ -20,19 +13,106 @@ type keyType = {
   
   @description('Expiry date in seconds since 1970-01-01T00:00:00Z.')
   attributesExp: int?
-
+  
   @description('Not before date in seconds since 1970-01-01T00:00:00Z.')
   attributesNbf: int?
-
+  
   @description('The elliptic curve name.')
-  curveName: curveNameType?
-
+  curveName: 'P-256' | 'P-256K' | 'P-384' | 'P-521'?
+  
   @description('The key size in bits. For example: 2048, 3072, or 4096 for RSA.')
   keySize: 2048 | 3072 | 4096 | null
 
   @description('The type of the key.')
-  kty: ktyType
+  kty: 'EC' | 'EC-HSM' | 'RSA' | 'RSA-HSM'
 
   @description('The key rotation policy.')
-  rotationPolicy: object?
+  rotationPolicy: {
+    @description('The attributes of the rotation policy.')
+    attributes: {
+      @description('The expiry time for the new key version. It should be in ISO8601 format. Eg: \'P90D\', \'P1Y\'.')
+      expiryTime: string
+    }
+
+    @description('The lifetime actions for the rotation policy.')
+    lifetimeActions: {
+      @description('The action to take.')
+      action: {
+        @description('The type of action to take.')
+        type: 'Rotate' | 'Notify'
+      }
+
+      @description('The trigger for the action.')
+      trigger: {
+        @description('The time before expiry to trigger the action.')
+        timeBeforeExpiry: string
+      }
+    }[]
+  }?
+
+  @description('The tags of the key vault (Optional).')
+  tags: object?
+
+  @description('The role assignments for the key vault (Optional).')
+  roleAssignments: roleAssignmentType?
```

Comment:
Put the array here and not in the type so it is more explicit that role assignments is a option array of roleAssignmentType